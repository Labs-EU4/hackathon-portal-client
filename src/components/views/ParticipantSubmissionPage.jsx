import React from "react";

import ParticipantSubmission from "../templates/ParticipantSubmission";

const ParticipantSubmissionPage = ({ id, setIsSubmitProjectOpen }) => (
    <ParticipantSubmission {...{id}} {...{setIsSubmitProjectOpen}} />
);

export default ParticipantSubmissionPage;