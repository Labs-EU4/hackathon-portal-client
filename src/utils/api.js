import Axios from "axios";
import { toast } from "react-toastify";
import { resetUser } from "../store/user/actions";

const baseURL = process.env.REACT_APP_API_URL;

// if process.env.CONTEXT = "production" {
//   Point to heroku API
// }
// if process.env.BRANCH = "develop" {
  // point to Heroku staging API
// }
// if process.env.CONTEXT = "deploy-preview" {
//   point to deploy-preview HEROKU API (the separate one)
// }



export const axiosWithAuth = token =>
  Axios.create({
    baseURL,
    headers: {
      Authorization: token
    }
  });

export const axios = Axios.create({ baseURL });

export const selectToken = state => state.currentUser.token;

export const showError = message => {
  toast.error(message);
};

export const showSuccess = message => {
  toast.success(message);
};

export function* handleError({ response }, put, history) {
  const { message, statusCode } = response.data;
  if (statusCode === 404) {
    history.push("/not-found");
  }
  if (message === "jwt expired") {
    yield put(resetUser());
  } else showError(`⚠️ ${message}`);
}
