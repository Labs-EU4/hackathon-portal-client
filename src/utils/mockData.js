export const initialState = {
  events: {
    data: [
      {
        id: 1,
        event_title: "EuroHack",
        event_description: "the best event hack hack hack macbook",
        creator_id: 1,
        start_date: "10/09/11",
        end_date: "08/07/13",
        location: "Europe",
        guidelines: "cool hekhkdsjfhsdkjf skdjfhskdjfhkjsdfh skjdfhksjdfh",
        participation_type: "Team",
        category_id: 1,
        requirements: [
          {
            video_url:
              "https://res.cloudinary.com/ibkrice-com/video/upload/v1581259780/Harvesting_ifato8.mp4"
          }
        ],
        ProfileImage: [
          {
            image_url:
              "https://res.cloudinary.com/ibkrice-com/video/upload/v1581259780/Harvesting_ifato8.mp4"
          }
        ],
        rubrics: [
          "Presentation",
          "Product Fit",
          "Functionality",
          "Innovation",
          "Product Design",
          "Extensibility"
        ]
      },
      {
        id: 2,
        event_title: "World",
        event_description: "number one hacker event in the world",
        creator_id: 1,
        start_date: "07/09/11",
        end_date: "08/07/13",
        location: "Asia",
        guidelines: "come well prepared,no cheating,no fighting",
        participation_type: "Team",
        category_id: 2
      }
    ],
    categories: [
      { id: 2, category_name: "Summer Hackton" },
      { id: 3, category_name: "World Hackton" },
      { id: 3, category_name: "Asia Hackton" },
      { id: 3, category_name: "Gamer's Hackton" },
      { id: 3, category_name: "Advanced Hackton" }
    ]
  },

  currentUser: {
    token: 347534574536,
    userId: 1,
    fullname: "Jake Tom",
    email: "8omemail@google.com",
    username: "Jake22",
    bio: "I like coding ",
    image: ""
  },
  submissions: [
    {
      id: 1,
      project_title: "Mock Project",
      participant_or_team_name: "Mock Team",
      event_id: 7,
      video_url: "",
      git_url: "https://www.google.com/",
      project_writeups: "No mock write up",
      submitted_by: 14,
      created_at: "2020-03-02",
      updated_at: "2020-03-02",
      average_rating: 0,
      acted_judges: 0,
      number_of_judges: 0
    }
  ],

  teams: [
    { id: 1, team_name: "The A team", team_lead: 2 },
    { id: 2, team_name: "Hacking central", team_lead: 3 },
    { id: 3, team_name: "blackhats", team_lead: 4 },
    { id: 4, team_name: "Spies", team_lead: 1 }
  ],
  participant_team_members: [
    {
      team_member: 3,
      team_id: 1
    },
    {
      team_member: 4,
      team_id: 1
    },
    {
      team_member: 5,
      team_id: 1
    }
  ]
};
