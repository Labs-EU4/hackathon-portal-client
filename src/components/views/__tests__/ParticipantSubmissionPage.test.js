import React from "react";
import { Router } from "react-router";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { render, cleanup } from "@testing-library/react";
import configureStore from "redux-mock-store";
import "@testing-library/jest-dom/extend-expect";
import ParticipantSubmissionPage from "../ParticipantSubmissionPage";
import { initialState } from "../../../utils/mockData";

const history = createMemoryHistory();

afterEach(cleanup);

let jestFeatures;
let mockStore;
let store;
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"), // use actual for all non-hook parts
  useParams: () => ({
    id: 1
  }),
  useRouteMatch: () => ({ url: "/dashboard/event/1/participant_submission" })
}));
beforeEach(() => {
  mockStore = configureStore();
  store = mockStore(initialState);
  jestFeatures = render(
    <Router history={history}>
      <Provider store={store}>
        <ParticipantSubmissionPage />
      </Provider>
    </Router>
  );
});

describe("Component ParticipantSubmissionPage.js text nodes renders properly", () => {
  it("asserts that the component renders properly", () => {
    expect(jestFeatures).toMatchSnapshot();
  });
  it("asserts that the right title text node for the event  renders properly", () => {
    expect(jestFeatures.getByText("EuroHack")).toMatchSnapshot();
  });
});
