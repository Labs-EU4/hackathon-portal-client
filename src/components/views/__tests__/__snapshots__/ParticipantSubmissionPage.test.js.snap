// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ParticipantSubmissionPage.js text nodes renders properly asserts that the component renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdVaJa sc-bwzfXH ckrujS"
      >
        <div
          class="sc-fjdhpX sc-cSHVUG kRWpKH"
        >
          <div
            class="sc-gzVnrw hdKmCx"
          >
            <h3
              class="sc-ifAKCX bZCfiG"
            >
              Submit Project
            </h3>
          </div>
          <form
            action="#"
          >
            <div
              class="sc-dnqmqq kaZCDV"
            >
              <p
                class="sc-kkGfuU eDlHfh"
              >
                You are making a submission for the
                 
                <strong>
                  EuroHack
                </strong>
                . Please ensure you have read the event guidelines and have gone through the grading rubrics for this event before you make your submission.
              </p>
            </div>
            <div
              class="sc-dnqmqq kaZCDV"
            >
              <div
                class="sc-gZMcBi ecCefW"
              >
                <label
                  class="sc-iAyFgw gHRRSC"
                  for="project_title"
                >
                  Project Title
                </label>
                <input
                  class="sc-jWBwVP fJzuhh"
                  display="wide"
                  id="project_title"
                  name="project_title"
                  type="text"
                  value=""
                />
                <span
                  class="sc-kpOJdX sc-eNQAEJ caiPaO"
                />
              </div>
              <div
                class="sc-gZMcBi ecCefW"
              >
                <label
                  class="sc-iAyFgw gHRRSC"
                  for="participant_or_team_nam"
                >
                  Team/Participant name
                </label>
                <input
                  class="sc-jWBwVP fJzuhh"
                  display="wide"
                  id="participant_or_team_name"
                  name="participant_or_team_name"
                  type="text"
                  value=""
                />
                <span
                  class="sc-kpOJdX sc-eNQAEJ caiPaO"
                />
              </div>
            </div>
            <div
              class="sc-dnqmqq bzpXAs"
            >
              <label
                class="sc-iAyFgw gHRRSC"
                for="project_writeups"
              >
                Project Writeup
              </label>
              <div
                class="sc-brqgnP huAjuz"
              >
                <textarea
                  id="project_writeups"
                  name="project_writeups"
                  type="text"
                />
              </div>
              <span
                class="sc-kpOJdX sc-eNQAEJ caiPaO"
              />
            </div>
            <div
              class="sc-dnqmqq kaZCDV"
            >
              <a
                class="sc-eHgmQL gOnppE"
                color="grey"
                href="/dashboard"
              >
                Cancel
              </a>
              <button
                class="sc-hSdWYo lgDWKZ"
                color="green"
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdVaJa sc-bwzfXH ckrujS"
    >
      <div
        class="sc-fjdhpX sc-cSHVUG kRWpKH"
      >
        <div
          class="sc-gzVnrw hdKmCx"
        >
          <h3
            class="sc-ifAKCX bZCfiG"
          >
            Submit Project
          </h3>
        </div>
        <form
          action="#"
        >
          <div
            class="sc-dnqmqq kaZCDV"
          >
            <p
              class="sc-kkGfuU eDlHfh"
            >
              You are making a submission for the
               
              <strong>
                EuroHack
              </strong>
              . Please ensure you have read the event guidelines and have gone through the grading rubrics for this event before you make your submission.
            </p>
          </div>
          <div
            class="sc-dnqmqq kaZCDV"
          >
            <div
              class="sc-gZMcBi ecCefW"
            >
              <label
                class="sc-iAyFgw gHRRSC"
                for="project_title"
              >
                Project Title
              </label>
              <input
                class="sc-jWBwVP fJzuhh"
                display="wide"
                id="project_title"
                name="project_title"
                type="text"
                value=""
              />
              <span
                class="sc-kpOJdX sc-eNQAEJ caiPaO"
              />
            </div>
            <div
              class="sc-gZMcBi ecCefW"
            >
              <label
                class="sc-iAyFgw gHRRSC"
                for="participant_or_team_nam"
              >
                Team/Participant name
              </label>
              <input
                class="sc-jWBwVP fJzuhh"
                display="wide"
                id="participant_or_team_name"
                name="participant_or_team_name"
                type="text"
                value=""
              />
              <span
                class="sc-kpOJdX sc-eNQAEJ caiPaO"
              />
            </div>
          </div>
          <div
            class="sc-dnqmqq bzpXAs"
          >
            <label
              class="sc-iAyFgw gHRRSC"
              for="project_writeups"
            >
              Project Writeup
            </label>
            <div
              class="sc-brqgnP huAjuz"
            >
              <textarea
                id="project_writeups"
                name="project_writeups"
                type="text"
              />
            </div>
            <span
              class="sc-kpOJdX sc-eNQAEJ caiPaO"
            />
          </div>
          <div
            class="sc-dnqmqq kaZCDV"
          >
            <a
              class="sc-eHgmQL gOnppE"
              color="grey"
              href="/dashboard"
            >
              Cancel
            </a>
            <button
              class="sc-hSdWYo lgDWKZ"
              color="green"
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
